chain-registry/src/relayer_config.rs

```diff
@@ -0,0 +1,239 @@
+/// Contains methods to generate a relayer config for a given chain
+use crate::{assetlist::AssetList, chain::ChainData, error::RegistryError};
+use futures::{stream::FuturesUnordered, StreamExt};
+use http::Uri;
+
+use ibc::core::ics24_host::identifier::ChainId;
+use ibc_proto::cosmos::bank::v1beta1::query_client::QueryClient;
+use ibc_relayer::config::types::{MaxMsgNum, MaxTxSize, Memo};
+use ibc_relayer::config::{default, filter::PacketFilter, AddressType, ChainConfig, GasPrice};
+use ibc_relayer::keyring::Store;
+use std::str::FromStr;
+
+use tendermint_light_client_verifier::types::TrustThreshold;
+use tendermint_rpc::{Client, HttpClient};
+use tokio;
+
+// ----------------- RPC ------------------
+
+/// Data retrievable from RPC endpoints
+#[derive(Clone, Debug)]
+pub struct RpcMandatoryData {
+    pub rpc_address: String,
+    pub max_block_size: u64,
+    // max_block_time should also be retrieved from the RPC
+    // however it looks like it is not in the genesis file anymore
+}
+
+/// Retrieves the RPC mandatory data from the RPC
+async fn query_rpc(rpc: &str) -> Result<RpcMandatoryData, RegistryError> {
+    let client =
+        HttpClient::new(rpc).map_err(|e| RegistryError::rpc_connect_error(rpc.to_string(), e))?;
+
+    let status = client
+        .status()
+        .await
+        .map_err(|e| RegistryError::rpc_status_error(rpc.to_string(), e))?;
+
+    if status.sync_info.catching_up {
+        return Err(RegistryError::rpc_syncing_error(rpc.to_string()));
+    }
+
+    let height = status.sync_info.latest_block_height;
+
+    match client.consensus_params(height).await {
+        Ok(response) => Ok(RpcMandatoryData {
+            rpc_address: rpc.to_string(),
+            max_block_size: response.consensus_params.block.max_bytes,
+        }),
+        Err(e) => Err(RegistryError::rpc_consensus_params_error(
+            rpc.to_string(),
+            e,
+        )),
+    }
+}
+
+/// Selects a healthy RPC from a list of RPCs
+/// An RPC is healthy if it can provide the RpcMandatoryData
+async fn select_healthy_rpc(rpcs: &[String]) -> Result<RpcMandatoryData, RegistryError> {
+    let mut futures = FuturesUnordered::new();
+    rpcs.iter().for_each(|rpc| {
+        futures.push(query_rpc(rpc));
+    });
```

Comment:
The use of iterators here could be a bit cleaner:
```suggestion
    let mut futures = rpcs
        .iter()
        .map(|rpc| query_rpc(rpc))
        .collect::<FuturesUnordered<_>>();
```